src/tests/suite/daffodil.test.ts

```diff
@@ -142,4 +150,138 @@ suite('Daffodfil', () => {
+        assert.strictEqual(
+          (await vscode.commands.getCommands()).includes(cmd),
+          false
+        )
+      })
+    })
+  })
+
+  suite('debug specifc commands', () => {
+    const debugSpecificCmds = [
+      'extension.dfdl-debug.toggleFormatting',
+      'hexview.display',
+      'infoset.display',
+      'infoset.diff',
+      'infoset.save',
+    ]
+
+    test('Not available by default', () => {
+      debugSpecificCmds.forEach(async (cmd) => {
+        assert.strictEqual(
+          (await vscode.commands.getCommands()).includes(cmd),
+          false
+        )
+      })
+    })
+
+    // This breaks when the omega-edit tests run for some reason
+    // test('Available when inDebugMode', () => {
+    //   vscode.commands.executeCommand('setContext', 'inDebugMode', true)
+
+    //   debugSpecificCmds.forEach(async (cmd) => {
+    //     assert.strictEqual(
+    //       (await vscode.commands.getCommands()).includes(cmd),
+    //       true
+    //     )
+    //   })
+    // })
+  })
+
+  suite('getCommands', () => {
+    test('getProgramName file exists', async () => {
+      assert.strictEqual(
+        await vscode.commands.executeCommand(
+          'extension.dfdl-debug.getProgramName',
+          testDfdlFile
+        ),
+        testDfdlFile
+      )
+    })
+
+    test('getProgramName file does not exists', async () => {
+      let file = path.join(__dirname, '../data/test.dfdl.xsd')
+
+      assert.notStrictEqual(
+        await vscode.commands.executeCommand(
+          'extension.dfdl-debug.getProgramName',
+          file
+        ),
+        file
+      )
+    })
+
+    test('getDataName file exists', async () => {
+      assert.strictEqual(
+        await vscode.commands.executeCommand(
+          'extension.dfdl-debug.getDataName',
+          testDfdlFile
+        ),
+        testDfdlFile
+      )
+    })
+
+    test('getDataName file does not exists', async () => {
+      let file = path.join(__dirname, '../data/test.dfdl.xsd')
+
+      assert.notStrictEqual(
+        await vscode.commands.executeCommand(
+          'extension.dfdl-debug.getDataName',
+          file
+        ),
+        file
+      )
+    })
+  })
+
+  suite('artifact attributes', () => {
+    const packageName = 'daffodil-debugger'
+    const packageVersion = '1.0.0'
+    const scriptName = 'daffodil-debugger'
+    const artifact = new Artifact(packageName, packageVersion, scriptName)
+
+    test('name set properly', () => {
+      assert.strictEqual(
+        artifact.name,
+        `${packageName}-${packageVersion}-${LIB_VERSION}`
+      )
+    })
+
+    test('archive set properly', () => {
+      assert.strictEqual(
+        artifact.archive,
+        `${packageName}-${packageVersion}-${LIB_VERSION}.zip`
+      )
+    })
+
+    test('scriptName set properly', () => {
+      assert.strictEqual(
+        artifact.scriptName,
+        os.platform() === 'win32' ? `${scriptName}.bat` : `./${scriptName}`
```

Comment:
Definitely make it a goal to use a library that suppresses all these platform conditionalizations before they proliferate. 

As people add new tests they should NOT copy this. 

We didn't do that soon enough in the daffodil-cli test rig, and the code-volume legacy of these sort of trivial conditionalization literring that test code is way too high.

Reply:
Created issue #301 to handle looking into this